examples/fungible-token/src/lib.rs

```diff
@@ -1,553 +1,155 @@
-    }
+near_contract_standards::impl_fungible_token_core!(Contract, token);
+near_contract_standards::impl_fungible_token_storage!(Contract, token);
 
-    /// Helper method to set the account details for `owner_id` to the state.
-    fn set_account(&mut self, owner_id: &AccountId, account: &Account) {
-        let account_hash = env::sha256(owner_id.as_bytes());
-        if account.balance > 0 || account.num_allowances > 0 {
-            self.accounts.insert(&account_hash, &account);
-        } else {
-            self.accounts.remove(&account_hash);
-        }
-    }
-
-    fn refund_storage(&self, initial_storage: StorageUsage) {
-        let current_storage = env::storage_usage();
-        let attached_deposit = env::attached_deposit();
-        let refund_amount = if current_storage > initial_storage {
-            let required_deposit =
-                Balance::from(current_storage - initial_storage) * STORAGE_PRICE_PER_BYTE;
-            assert!(
-                required_deposit <= attached_deposit,
-                "The required attached deposit is {}, but the given attached deposit is is {}",
-                required_deposit,
-                attached_deposit,
-            );
-            attached_deposit - required_deposit
-        } else {
-            attached_deposit
-                + Balance::from(initial_storage - current_storage) * STORAGE_PRICE_PER_BYTE
-        };
-        if refund_amount > 0 {
-            env::log(format!("Refunding {} tokens for storage", refund_amount).as_bytes());
-            Promise::new(env::predecessor_account_id()).transfer(refund_amount);
+#[near_bindgen]
+impl FungibleTokenMetadataProvider for Contract {
+    fn ft_metadata(&self) -> FungibleTokenMetadata {
+        FungibleTokenMetadata {
+            spec: FT_METADATA_VERSION.to_string(),
+            name: "Example NEAR fungible token".to_string(),
+            symbol: "EXAMPLE".to_string(),
+            icon: Some(
+                "https://near.org/wp-content/themes/near-19/assets/img/brand-icon.png".to_string(),
+            ),
+            reference: self.reference.clone(),
+            reference_hash: self.reference_hash,
+            decimals: 24,
         }
     }
 }
 
 #[cfg(all(test, not(target_arch = "wasm32")))]
 mod tests {
+    use near_sdk::test_utils::{accounts, VMContextBuilder};
     use near_sdk::MockedBlockchain;
-    use near_sdk::{testing_env, VMContext};
+    use near_sdk::{testing_env, Balance};
 
     use super::*;
+    use std::convert::TryInto;
 
-    fn alice() -> AccountId {
-        "alice.near".to_string()
-    }
-    fn bob() -> AccountId {
-        "bob.near".to_string()
-    }
-    fn carol() -> AccountId {
-        "carol.near".to_string()
-    }
+    const REFERENCE: &str =
+        "https://github.com/near/near-sdk-rs/tree/master/examples/fungible-token";
+    const REFERENCE_HASH: &str = "Aa4hsn9vdMetr2WDvYWduLCFpi6VZqJ3AzDm16VmSibG";
```

Comment:
How is this computed?  Is this the commit hash?

Reply:
It's just a random hash. Was replaced with obscure `vec[1; 32].into()`